tests/test_core.py

```diff
@@ -57,12 +57,94 @@ class test_Variable(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different bounds."""
-        raise NotImplementedError()
+        v = relentless.core.Variable(value=1.0, const=True)
+        self.assertAlmostEqual(v.value, 1.0)
+        self.assertEqual(v.const, True)
+        self.assertEqual(v.low, None)
+        self.assertEqual(v.high, None)
+        self.assertEqual(v.isfree(), True)
+        self.assertEqual(v.atlow(), False)
+        self.assertEqual(v.athigh(), False)
+
+        v = relentless.core.Variable(value=1.2, low=0.0, high=2.0)
+        self.assertAlmostEqual(v.value, 1.2)
+        self.assertAlmostEqual(v.low, 0.0)
+        self.assertAlmostEqual(v.high, 2.0)
+        self.assertEqual(v.isfree(), True)
+        self.assertEqual(v.atlow(), False)
+        self.assertEqual(v.athigh(), False)
+
+        v = relentless.core.Variable(value=-1, low=0.5)
+        self.assertAlmostEqual(v.value, 0.5)
+        self.assertAlmostEqual(v.low, 0.5)
+        self.assertEqual(v.high, None)
+        self.assertEqual(v.isfree(), False)
+        self.assertEqual(v.atlow(), True)
+        self.assertEqual(v.athigh(), False)
+
+        v = relentless.core.Variable(value=2.2, high=2.0)
+        self.assertAlmostEqual(v.value, 2.0)
+        self.assertEqual(v.high, 2.0)
+        self.assertEqual(v.low, None)
+        self.assertEqual(v.isfree(), False)
+        self.assertEqual(v.atlow(), False)
+        self.assertEqual(v.athigh(), True)
+
+        with self.assertRaises(ValueError):
+            v = relentless.core.Variable(value='4')
 
     def test_clamp(self):
         """Test methods for clamping values with bounds."""
-        raise NotImplementedError()
+        v = relentless.core.Variable(value=0.0, low=2.0)
```

Comment:
To make these tests easier to read later, can you add a comment before each block of tests? Like
```py
# clamp with lower bound
...

# clamp with lower and upper bound
...

# clamp with no bounds
```